### PR TITLE
Koa 4814/update to ios 15

### PR DESCRIPTION
**Pull request by frugoman**
Updating the project to compile using iOS 15 and Xcode 13.1

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [ ] `README.md`
+ [x] Tests
+ [ ] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [ ] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)
+ [ ] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -97,7 +97,7 @@ end
 desc "Run unit tests up to #{MAX_TEST_REPEATS} times until they pass"
 task :test do
   only_testing = FULL_TESTS ? '' : '-only-testing:Backpack_Tests'
-  test_command = "set -o pipefail && xcodebuild test -enableCodeCoverage YES -workspace #{EXAMPLE_WORKSPACE} -scheme \"#{EXAMPLE_SCHEMA}\" SWIFT_VERSION=#{SWIFT} -sdk #{BUILD_SDK} -destination \"#{DESTINATION}\" #{only_testing} ONLY_ACTIVE_ARCH=YES | xcpretty"
+  test_command = "set -o pipefail && xcodebuild test -enableCodeCoverage YES -workspace #{EXAMPLE_WORKSPACE} -scheme \"#{EXAMPLE_SCHEMA}\" SWIFT_VERSION=#{SWIFT} -sdk #{BUILD_SDK} -destination \"#{DESTINATION}\" #{only_testing} ONLY_ACTIVE_ARCH=YES | xcpretty | grep -v Pods/TTTAttributedLabel/TTTAttributedLabel/ | grep -v Pods/OCMock/ | grep -v Pods/MBProgressHUD/ | grep -v Pods/FloatingPanel/ | grep -A 5 \"#{ANALYZE_FAIL_MESSAGE}\"
```
Comment (zgeor) [inline Rakefile:100]: Previously this was in the build-time error and warnings task, why did these move ?

Comment (zgeor) [reply Rakefile:100]: Also both of these should be creating an html report with xcpretty, so we can export it with GHActions

Comment (frugoman) [reply Rakefile:100]: good catch, my bad moving code around.

File: Rakefile [hunk 2]
```diff
@@ -86,7 +86,7 @@ end
 
 desc "Check for build-time errors and warnings"
 task :analyze do
-  sh "set -o pipefail && ! xcodebuild -workspace #{EXAMPLE_WORKSPACE} -scheme \"#{EXAMPLE_SCHEMA}\" SWIFT_VERSION=#{SWIFT} -sdk #{BUILD_SDK} -destination \"#{DESTINATION}\" ONLY_ACTIVE_ARCH=YES analyze 2>&1 | xcpretty | grep -v Pods/TTTAttributedLabel/TTTAttributedLabel/ | grep -v Pods/OCMock/ | grep -v Pods/MBProgressHUD/ | grep -v Pods/FloatingPanel/ | grep -A 5 \"#{ANALYZE_FAIL_MESSAGE}\""
```
Comment (frugoman) [inline Rakefile:89]: this one's needed, analysis was succeeding, but the `!` was inverting the result and failing as a consequence.

**Review by zgeor — APPROVED** [review]

**Review by olliecurtis — APPROVED** [review]

**Pull request merged**